src/Symfony/Component/Config/Definition/PrototypedArrayNode.php

```diff
@@ -225,7 +225,7 @@ protected function normalizeValue($value)
 
         $value = $this->remapXml($value);
 
-        $isAssoc = array_keys($value) !== range(0, \count($value) - 1);
+        $isAssoc = !array_is_list($value);
```

Comment:
maybe `$isList = array_is_list($value);` to prevent the double negation? (same in the other method)